web/packages/teleport/src/Bots/Details/BotDetails.test.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { QueryClientProvider } from '@tanstack/react-query';
+import { setupServer } from 'msw/node';
+import { PropsWithChildren } from 'react';
+import { MemoryRouter, useHistory } from 'react-router';
+
+import darkTheme from 'design/theme/themes/darkTheme';
+import { ConfiguredThemeProvider } from 'design/ThemeProvider';
+import {
+  fireEvent,
+  render,
+  screen,
+  testQueryClient,
+  waitForElementToBeRemoved,
+  within,
+} from 'design/utils/testing';
+import { InfoGuidePanelProvider } from 'shared/components/SlidingSidePanel/InfoGuide';
+
+import { ContextProvider } from 'teleport/index';
+import { createTeleportContext } from 'teleport/mocks/contexts';
+import { defaultAccess, makeAcl } from 'teleport/services/user/makeAcl';
+import { getBotError, getBotSuccess } from 'teleport/test/helpers/bots';
+
+import { BotDetails } from './BotDetails';
+
+jest.mock('react-router', () => {
+  const actual = jest.requireActual('react-router');
+  return {
+    ...actual,
+    useHistory: jest.fn(),
+    useParams: jest.fn(() => ({
+      name: 'test-bot-name',
+    })),
+  };
+});
```

Comment:
I don't have much experience with writing tests that use React Router, but I haven't seen other people mock the router like this. Would it be possible to pass `initialEntries` to `<MemoryRouter>` instead? That's what I typically found in our tests and stories.